backend/danswer/server/settings/api.py

```diff
@@ -99,21 +100,32 @@ def get_user_notifications(
         logger.warning("Could not find reindex flag")
         return []
 
-    reindex_notifs = get_notifications(
-        user=user, notif_type=NotificationType.REINDEX, db_session=db_session
-    )
-
-    if not reindex_notifs:
-        notif = create_notification(
-            user=user, notif_type=NotificationType.REINDEX, db_session=db_session
-        )
-        return [Notification.from_model(notif)]
-
-    if len(reindex_notifs) > 1:
-        logger.error("User has multiple reindex notifications")
-
-    reindex_notif = reindex_notifs[0]
+    try:
+        with db_session.begin_nested():  # Ensure we don't  overcount the reindex notifications
```

Comment:
Would add a slightly more explicit comment here on how explicitly tying the read operations here to the transaction so that they can run extremely close in time to the write (as it's passed to the DB together rather than in 2 calls).

Also would make this just begin transaction instead of nested.